src/Compilers/CSharp/Portable/CodeGen/EmitArrayInitializer.cs

```diff
@@ -469,5 +476,43 @@ private int TryGetRawDataForArrayInit(BoundArrayInitialization initializer, out
             data = writer.ToImmutableArray();
             return elementCount;
         }
+
+        /// <summary>
+        ///  Returns a byte blob that matches serialized content of zero-valued array.
+        /// </summary>
+        private int TryGetRawDataForDefaultArray(BoundArrayCreation initializer, out ImmutableArray<byte> data)
+        {
+            data = default;
+
+            if (initializer.Bounds.Length != 1)
+            {
+                return -1;
+            }
+
+            var elementCountValue = initializer.Bounds[0].ConstantValue;
+            if (elementCountValue == null)
+            {
+                return -1;
+            }
+
+            var elementCount = elementCountValue.Int32Value;
+            if (elementCount == 0)
+            {
+                data = ImmutableArray<byte>.Empty;
+                return 0;
+            }
+
+            var arrayType = (ArrayTypeSymbol)initializer.Type;
+            var elementType = arrayType.ElementType.SpecialType;
+            var constantValue = ConstantValue.Default(elementType);
+            var writer = new BlobBuilder(elementCount * 4);
```

Comment:
I initialized the builder in the same way as it's done in the method above but why do we need here `* 4`? Should it be something like `elementCount * sizeof(element)`?